# Compet - Formação 4.0

Aluna: Ruth Xavier

### Carregando o conjunto de dados

**Base escolhida:** Adequação da Formação Docente (2024)

**Descrição:** Percentual de docentes na Educação Infantil (Creche e Pré-Escola), no Ensino Fundamental, no Ensino Médio e Educação de Jovens e Adultos por grupo de adequação da formação à disciplina que leciona, segundo localização e dependência administrativa - Brasil, Regiões Geográficas e Unidades da Federação - 2024

#### Dicionário dos dados: 

01. NU_ANO_CENSO: Ano em que foi realizado o censo;
02. UNIDGEO: Unidade Geográfica; 
03. NO_CATEGORIA: Categorizando em zona Rural, Urbana e Total; 
04. NO_DEPENDENCIA: Dependência administrativa; 
05. ED_INF_CAT_1: Educação Infantil Grupo 1; 
06. ED_INF_CAT_2: Educação Infantil Grupo 2; 
07. ED_INF_CAT_3: Educação Infantil Grupo 3; 
08. ED_INF_CAT_4: Educação Infantil Grupo 5;
09. FUN_CAT_1: Ensino Fundamental Grupo 1 total;  
10. FUN_CAT_2: Ensino Fundamental Grupo 2 total;
11. FUN_CAT_3: Ensino Fundamental Grupo 3 total; 
12. FUN_CAT_4: Ensino Fundamental Grupo 4 total;  
13. FUN_CAT_5: Ensino Fundamental Grupo 5 total; 
14. FUN_AI_CAT_1: Ensino Fundamental Grupo 1 - Anos Iniciais; 
15. FUN_AI_CAT_2: Ensino Fundamental Grupo 2 - Anos Iniciais; 
16. FUN_AI_CAT_3: Ensino Fundamental Grupo 3 - Anos Iniciais; 
17. FUN_AI_CAT_4: Ensino Fundamental Grupo 4 - Anos Iniciais; 
18. FUN_AI_CAT_5: Ensino Fundamental Grupo 5 - Anos Iniciais; 
19. FUN_AF_CAT_1: Ensino Fundamental Grupo 1 - Anos Finais;
20. FUN_AF_CAT_2: Ensino Fundamental Grupo 2 - Anos Finais;
21. FUN_AF_CAT_3: Ensino Fundamental Grupo 3 - Anos Finais;
22. FUN_AF_CAT_4: Ensino Fundamental Grupo 4 - Anos Finais;
23. FUN_AF_CAT_5: Ensino Fundamental Grupo 5 - Anos Finais;
24. MED_CAT_1: Ensino Médio Grupo 1; 
25. MED_CAT_2: Ensino Médio Grupo 2;
26. MED_CAT_3: Ensino Médio Grupo 3;
27. MED_CAT_4: Ensino Médio Grupo 4;
28. MED_CAT_5: Ensino Médio Grupo 5;
29. EJA_FUN_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Fundamental;
30. EJA_FUN_CAT_2: Educação de Jovens e Adultos Grupo 2 - Ensino Fundamental;
31. EJA_FUN_CAT_3: Educação de Jovens e Adultos Grupo 3 - Ensino Fundamental;
32. EJA_FUN_CAT_4: Educação de Jovens e Adultos Grupo 4 - Ensino Fundamental;
33. EJA_FUN_CAT_5: Educação de Jovens e Adultos Grupo 5 - Ensino Fundamental;
34. EJA_MED_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Médio;
35. EJA_MED_CAT_2: Educação de Jovens e Adultos Grupo 2 - Ensino Médio;
36. EJA_MED_CAT_3: Educação de Jovens e Adultos Grupo 3 - Ensino Médio;
37. EJA_MED_CAT_4: Educação de Jovens e Adultos Grupo 4 - Ensino Médio;
38. EJA_MED_CAT_5: Educação de Jovens e Adultos Grupo 5 - Ensino Médio;
39. EJA_MED_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Médio;

GRUPO 1: Docentes com formação superior de licenciatura na mesma disciplina que lecionam, ou bacharelado na mesma disciplina com curso de complementação pedagógica concluído;

GRUPO 2: Docentes com formação superior de bacharelado na disciplina
correspondente, mas sem licenciatura ou complementação pedagógica;

GRUPO 3: Docentes com licenciatura em área diferente daquela que lecionam, ou com bacharelado nas disciplinas da base curricular comum e complementação pedagógica concluída em área diferente daquela que lecionam;

GRUPO 4: Docentes com outra formação superior não considerada nas categorias anteriores;

GRUPO 5: Docentes que não possuem curso superior completo.

In [30]:
# Importando bibliotecas

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns 
from sklearn.impute import SimpleImputer

In [5]:
afd = pd.read_excel('datasets/AFD_BRASIL_REGIOES_UFS_2024.xlsx', skiprows=10)

In [6]:
afd.head()

NU_ANO_CENSO UNIDGEO NO_CATEGORIA NO_DEPENDENCIA ED_INF_CAT_1 ED_INF_CAT_2  \
0         2024  Brasil        Total          Total         64.1          0.7   
1         2024  Brasil       Urbana          Total         64.7          0.7   
2         2024  Brasil        Rural          Total         57.9          0.9   
3         2024  Brasil        Total        Federal         56.4          2.3   
4         2024  Brasil       Urbana        Federal         55.8          2.4   

  ED_INF_CAT_3 ED_INF_CAT_4 ED_INF_CAT_5 FUN_CAT_1  ... EJA_FUN_CAT_1  \
0         17.8          1.8         15.6      68.8  ...            27   
1         18.2          1.9         14.5      73.3  ...          35.5   
2         14.1          1.7         25.4      46.6  ...            10   
3         26.3          7.7          7.3      87.2  ...          68.9   
4         26.7          7.9          7.2      87.2  ...          68.9   

  EJA_FUN_CAT_2 EJA_FUN_CAT_3 EJA_FUN_CAT_4 EJA_FUN_CAT_5 EJA_MED_CAT_1  \
0           0.5          44.6           2.4          25.5          56.1   
1           0.7          47.9           2.7          13.2          57.9   
2           0.1          38.1           1.8            50          38.4   
3             6          17.5           5.5           2.1          83.3   
4             6          17.5           5.5           2.1          83.2   

  EJA_MED_CAT_2 EJA_MED_CAT_3 EJA_MED_CAT_4 EJA_MED_CAT_5  
0           1.1          36.1           3.1           3.6  
1           1.2          35.1           3.1           2.7  
2           0.6            46           3.6          11.4  
3           6.7           6.4           2.9           0.7  
4           6.8           6.3             3           0.7  

[5 rows x 39 columns]

In [7]:
afd.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 39 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   NU_ANO_CENSO    594 non-null    object
 1   UNIDGEO         588 non-null    object
 2   NO_CATEGORIA    588 non-null    object
 3   NO_DEPENDENCIA  588 non-null    object
 4   ED_INF_CAT_1    588 non-null    object
 5   ED_INF_CAT_2    588 non-null    object
 6   ED_INF_CAT_3    588 non-null    object
 7   ED_INF_CAT_4    588 non-null    object
 8   ED_INF_CAT_5    588 non-null    object
 9   FUN_CAT_1       588 non-null    object
 10  FUN_CAT_2       588 non-null    object
 11  FUN_CAT_3       588 non-null    object
 12  FUN_CAT_4       588 non-null    object
 13  FUN_CAT_5       588 non-null    object
 14  FUN_AI_CAT_1    588 non-null    object
 15  FUN_AI_CAT_2    588 non-null    object
 16  FUN_AI_CAT_3    588 non-null    object
 17  FUN_AI_CAT_4    588 non-null    object
 18  FUN_AI_CAT

In [ ]:
# Verificando a existência de dados nulos

afd.isna().any().any()

np.True_

In [ ]:
#Observando o total de nulos por coluna

afd.isna().sum()

NU_ANO_CENSO      1
UNIDGEO           7
NO_CATEGORIA      7
NO_DEPENDENCIA    7
ED_INF_CAT_1      7
ED_INF_CAT_2      7
ED_INF_CAT_3      7
ED_INF_CAT_4      7
ED_INF_CAT_5      7
FUN_CAT_1         7
FUN_CAT_2         7
FUN_CAT_3         7
FUN_CAT_4         7
FUN_CAT_5         7
FUN_AI_CAT_1      7
FUN_AI_CAT_2      7
FUN_AI_CAT_3      7
FUN_AI_CAT_4      7
FUN_AI_CAT_5      7
FUN_AF_CAT_1      7
FUN_AF_CAT_2      7
FUN_AF_CAT_3      7
FUN_AF_CAT_4      7
FUN_AF_CAT_5      7
MED_CAT_1         7
MED_CAT_2         7
MED_CAT_3         7
MED_CAT_4         7
MED_CAT_5         7
EJA_FUN_CAT_1     7
EJA_FUN_CAT_2     7
EJA_FUN_CAT_3     7
EJA_FUN_CAT_4     7
EJA_FUN_CAT_5     7
EJA_MED_CAT_1     7
EJA_MED_CAT_2     7
EJA_MED_CAT_3     7
EJA_MED_CAT_4     7
EJA_MED_CAT_5     7
dtype: int64

Através da visualização dos dados com afd.head() e afd.info() e afd.isna().any().any() foi possível identificar que: 

- Há 39 colunas e 595 linhas; 
- A maioria das colunas possuem 588 valores que não são linhas, logo 6 linhas estão vazias (devido ao skiprow, geralmente exibe uma linha a mais no final); 
- E todas as colunas foram identificadas como object, pois provavelmente há valores não numéricos como o uso do "--" ou que estejam vazias.

## Preparação dos dados para análise

A base de dados referesse a uma coleta de dados realizada em 2024 a cerca do indiciador educacional a cerca da adequação da formação docente, logo isso significa que essa base tem por objetivo mapear se os docentes tem uma formação em conformidade entre a disciplina que lecionam e suas respectivas formações acadêmicas. 

De acordo com a LDB, Lei de Diretrizes e Base da Educação, Lei nº9.394/1996, no artigo 62, que contempla os dados presentes:

    Art. 62. A formação de docentes para atuar na educação básica far-se-á em nível superior, em curso de licenciatura, de graduação plena, em universidades e institutos superiores de educação, admitida, como formação mínima para o exercício do magistério na educação infantil e nos 5 (cinco) primeiros anos do ensino fundamental, a oferecida em nível médio na modalidade normal.

**Perguntas para os dados:**

Primeiramente, é importante definir um foco principal: Educação de Jovens e Adultos - Ensino Fundamental e Ensino Médio

*Educação de Jovens e Adultos* 

1. Em qual zona a EJA sofre uma maior discrepância entre a formação docente e a disciplina ensinada num total?

2. Analisando separadamente, qual zona apresenta maior discrepância na formação docente para a EJA do Ensino Fundamental? E para a EJA do Ensino Médio?

3. Em quais dependências admnistritativas há uma maior incidênica dessa discrepância em relação a EJA do Ensino Fundamental? E no Ensino Médio? 

4. Há alguma relação entre a unidade geográfica e a localização em realação a EJA do Ensino Fundamental?  E no Ensino Médio? 

5. Existe alguma correlação entre a unidade geográfica e a dependencia administrativa em relação à formação docente da EJA no Ensino Fundamental? E no Ensino Médio?

6. Ao fazer uma comparação entre o Ensino Fundamental EJA e Ensino Médio EJA, qual podemos observar que há uma maior dispariadade?

7. Qual região do Brasil apresenta a maior variação na formação dos docentes da EJA em termos de adequação (Grupos 1 a 5)?

8. Qual grupo de formação docente (1, 2, 3, 4 e 5) concentra um maior percentual desses docentes, considerando a zona geográfica?

9. Qual grupo de formação docente (Grupo 1, 2, 3, 4, 5) apresenta maior percentual de docentes na EJA, considerando a dependência administrativa?

10. Comparando as regiões do Brasil, qual apresenta maior desigualdade na adequação da formação docente para a EJA?


### Selecionando dados  

As colunas selecionadas para a análise:

*possíveis filtros* 
UNIDGEO, NO_CATEGORIA, NO_DEPENDENCIA

*Etapa de ensino - Educação de Jovens e Adultos - Ensino Fundamental*
EJA_FUN_CAT_1, EJA_FUN_CAT_2, EJA_FUN_CAT_3, EJA_FUN_CAT_4, EJA_FUN_CAT_5

*Etapa de ensino - Educação de Jovens e Adultos - Ensino Médio*
EJA_MED_CAT_1,EJA_MED_CAT_2, EJA_MED_CAT_3, EJA_MED_CAT_4, EJA_MED_CAT_5


In [10]:
afd_eja = afd.iloc[:, [1, 2, 3, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38]]

afd_eja

UNIDGEO NO_CATEGORIA NO_DEPENDENCIA EJA_FUN_CAT_1 EJA_FUN_CAT_2  \
0    Brasil        Total          Total            27           0.5   
1    Brasil       Urbana          Total          35.5           0.7   
2    Brasil        Rural          Total            10           0.1   
3    Brasil        Total        Federal          68.9             6   
4    Brasil       Urbana        Federal          68.9             6   
..      ...          ...            ...           ...           ...   
590     NaN          NaN            NaN           NaN           NaN   
591     NaN          NaN            NaN           NaN           NaN   
592     NaN          NaN            NaN           NaN           NaN   
593     NaN          NaN            NaN           NaN           NaN   
594     NaN          NaN            NaN           NaN           NaN   

    EJA_FUN_CAT_3 EJA_FUN_CAT_4 EJA_FUN_CAT_5 EJA_MED_CAT_1 EJA_MED_CAT_2  \
0            44.6           2.4          25.5          56.1           1.1   
1            47.9           2.7          13.2          57.9           1.2   
2            38.1           1.8            50          38.4           0.6   
3            17.5           5.5           2.1          83.3           6.7   
4            17.5           5.5           2.1          83.2           6.8   
..            ...           ...           ...           ...           ...   
590           NaN           NaN           NaN           NaN           NaN   
591           NaN           NaN           NaN           NaN           NaN   
592           NaN           NaN           NaN           NaN           NaN   
593           NaN           NaN           NaN           NaN           NaN   
594           NaN           NaN           NaN           NaN           NaN   

    EJA_MED_CAT_3 EJA_MED_CAT_4 EJA_MED_CAT_5  
0            36.1           3.1           3.6  
1            35.1           3.1           2.7  
2              46           3.6          11.4  
3             6.4           2.9           0.7  
4             6.3             3           0.7  
..            ...           ...           ...  
590           NaN           NaN           NaN  
591           NaN           NaN           NaN  
592           NaN           NaN           NaN  
593           NaN           NaN           NaN  
594           NaN           NaN           NaN  

[595 rows x 13 columns]

In [11]:
# Dicionários com as colunas selecionadas para análise

data_dictionary = pd.DataFrame([
    {
        'Variável': 'GEO',
        'Descrição': 'Unidade Geográfica',
        'Tipo': 'Categórico', 
        'Subtipo': 'Nominal',
        'Obs': 'Região do Brasil'
    },
    {
        'Variável': 'LOC',
        'Descrição': 'Categoria',
        'Tipo': 'Categórico', 
        'Subtipo': 'Nominal',
        'Obs': 'Zona Rural e Urbana'
    },
    {
        'Variável': 'DA',
        'Descrição': 'Dependência Administrativa',
        'Tipo': 'Categórico', 
        'Subtipo': 'Ordinal',
        'Obs': 'Municipal, Estudal, Federal e Privado'
    },
    {
        'Variável': 'EJAFUNG1',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 1',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG2',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 2',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG3',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 3',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG4',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 4',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG5',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 5',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG1',
        'Descrição': 'EJA - Ensino Médio - Grupo 1',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG2',
        'Descrição': 'EJA - Ensino Médio - Grupo 2',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG3',
        'Descrição': 'EJA - Ensino Médio - Grupo 3',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG4',
        'Descrição': 'EJA - Ensino Médio - Grupo 4',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG5',
        'Descrição': 'EJA - Ensino Médio - Grupo 5',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    }
])

In [12]:
data_dictionary

Variável                           Descrição        Tipo   Subtipo  \
0        GEO                  Unidade Geográfica  Categórico   Nominal   
1        LOC                           Categoria  Categórico   Nominal   
2         DA          Dependência Administrativa  Categórico   Ordinal   
3   EJAFUNG1  EJA - Ensino Fundamental - Grupo 1    Numérico  Contínuo   
4   EJAFUNG2  EJA - Ensino Fundamental - Grupo 2    Numérico  Contínuo   
5   EJAFUNG3  EJA - Ensino Fundamental - Grupo 3    Numérico  Contínuo   
6   EJAFUNG4  EJA - Ensino Fundamental - Grupo 4    Numérico  Contínuo   
7   EJAFUNG5  EJA - Ensino Fundamental - Grupo 5    Numérico  Contínuo   
8   EJAMEDG1        EJA - Ensino Médio - Grupo 1    Numérico  Contínuo   
9   EJAMEDG2        EJA - Ensino Médio - Grupo 2    Numérico  Contínuo   
10  EJAMEDG3        EJA - Ensino Médio - Grupo 3    Numérico  Contínuo   
11  EJAMEDG4        EJA - Ensino Médio - Grupo 4    Numérico  Contínuo   
12  EJAMEDG5        EJA - Ensino Médio - Grupo 5    Numérico  Contínuo   

                                      Obs  
0                        Região do Brasil  
1                     Zona Rural e Urbana  
2   Municipal, Estudal, Federal e Privado  
3                                          
4                                          
5                                          
6                                          
7                                          
8                                          
9                                          
10                                         
11                                         
12

In [ ]:
# Renomeando as colunas 

colunas = ["GEO", "LOC", "DA", "EJAFUNG1", "EJAFUNG2", "EJAFUNG3", "EJAFUNG4", "EJAFUNG5", "EJAMED1", "EJAMED2", "EJAMED3", "EJAMED4", "EJAMED5"]
afd_eja.columns = colunas

afd_eja


GEO     LOC       DA EJAFUNG1 EJAFUNG2 EJAFUNG3 EJAFUNG4 EJAFUNG5  \
0    Brasil   Total    Total       27      0.5     44.6      2.4     25.5   
1    Brasil  Urbana    Total     35.5      0.7     47.9      2.7     13.2   
2    Brasil   Rural    Total       10      0.1     38.1      1.8       50   
3    Brasil   Total  Federal     68.9        6     17.5      5.5      2.1   
4    Brasil  Urbana  Federal     68.9        6     17.5      5.5      2.1   
..      ...     ...      ...      ...      ...      ...      ...      ...   
590     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
591     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
592     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
593     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
594     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   

    EJAMED1 EJAMED2 EJAMED3 EJAMED4 EJAMED5  
0      56.1     1.1    36.1     3.1     3.6  
1      57.9     1.2    35.1     3.1     2.7  
2      38.4     0.6      46     3.6    11.4  
3      83.3     6.7     6.4     2.9     0.7  
4      83.2     6.8     6.3       3     0.7  
..      ...     ...     ...     ...     ...  
590     NaN     NaN     NaN     NaN     NaN  
591     NaN     NaN     NaN     NaN     NaN  
592     NaN     NaN     NaN     NaN     NaN  
593     NaN     NaN     NaN     NaN     NaN  
594     NaN     NaN     NaN     NaN     NaN  

[595 rows x 13 columns]

In [14]:
afd_eja.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   GEO       588 non-null    object
 1   LOC       588 non-null    object
 2   DA        588 non-null    object
 3   EJAFUNG1  588 non-null    object
 4   EJAFUNG2  588 non-null    object
 5   EJAFUNG3  588 non-null    object
 6   EJAFUNG4  588 non-null    object
 7   EJAFUNG5  588 non-null    object
 8   EJAMED1   588 non-null    object
 9   EJAMED2   588 non-null    object
 10  EJAMED3   588 non-null    object
 11  EJAMED4   588 non-null    object
 12  EJAMED5   588 non-null    object
dtypes: object(13)
memory usage: 60.6+ KB


É possível perceber os valores ausentes nessas 13 colunas seguem sendo identificadas como object. Portanto, para resolver esse problema irei substituir esses valores ausentes resolvi optar pela técnica de imputação Mediana, pela sua robustêz e por ser pouco afetada pelos outliers que podem sugir se eu usasse a média. 

In [ ]:
# Fazendo uma cópia do meu df original

df = afd_eja.copy()

In [18]:
# Iniciando a substituição de "--" para NaN nas colunas com dados numéricos

col_num = ['EJAFUNG1', 'EJAFUNG2', 'EJAFUNG3', 'EJAFUNG4', 'EJAFUNG5', 'EJAMED1', 'EJAMED2', 'EJAMED3', 'EJAMED4', 'EJAMED5'] #minhas colunas numéricas

In [22]:
#Substituindo "--" por nan apenas nessas colunas

afd_eja[col_num] = afd_eja[col_num].replace('--', np.nan) 

C:\Users\ruthm\AppData\Local\Temp\ipykernel_19492\3105151559.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  afd_eja[col_num] = afd_eja[col_num].replace('--', np.nan)


In [23]:
afd_eja

GEO     LOC       DA  EJAFUNG1  EJAFUNG2  EJAFUNG3  EJAFUNG4  \
0    Brasil   Total    Total      27.0       0.5      44.6       2.4   
1    Brasil  Urbana    Total      35.5       0.7      47.9       2.7   
2    Brasil   Rural    Total      10.0       0.1      38.1       1.8   
3    Brasil   Total  Federal      68.9       6.0      17.5       5.5   
4    Brasil  Urbana  Federal      68.9       6.0      17.5       5.5   
..      ...     ...      ...       ...       ...       ...       ...   
590     NaN     NaN      NaN       NaN       NaN       NaN       NaN   
591     NaN     NaN      NaN       NaN       NaN       NaN       NaN   
592     NaN     NaN      NaN       NaN       NaN       NaN       NaN   
593     NaN     NaN      NaN       NaN       NaN       NaN       NaN   
594     NaN     NaN      NaN       NaN       NaN       NaN       NaN   

     EJAFUNG5  EJAMED1  EJAMED2  EJAMED3  EJAMED4  EJAMED5  
0        25.5     56.1      1.1     36.1      3.1      3.6  
1        13.2     57.9      1.2     35.1      3.1      2.7  
2        50.0     38.4      0.6     46.0      3.6     11.4  
3         2.1     83.3      6.7      6.4      2.9      0.7  
4         2.1     83.2      6.8      6.3      3.0      0.7  
..        ...      ...      ...      ...      ...      ...  
590       NaN      NaN      NaN      NaN      NaN      NaN  
591       NaN      NaN      NaN      NaN      NaN      NaN  
592       NaN      NaN      NaN      NaN      NaN      NaN  
593       NaN      NaN      NaN      NaN      NaN      NaN  
594       NaN      NaN      NaN      NaN      NaN      NaN  

[595 rows x 13 columns]

In [ ]:
afd_eja.info() # as colunas já não estão mais todas tipo object

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   GEO       588 non-null    object 
 1   LOC       588 non-null    object 
 2   DA        588 non-null    object 
 3   EJAFUNG1  489 non-null    float64
 4   EJAFUNG2  489 non-null    float64
 5   EJAFUNG3  489 non-null    float64
 6   EJAFUNG4  489 non-null    float64
 7   EJAFUNG5  489 non-null    float64
 8   EJAMED1   508 non-null    float64
 9   EJAMED2   508 non-null    float64
 10  EJAMED3   508 non-null    float64
 11  EJAMED4   508 non-null    float64
 12  EJAMED5   508 non-null    float64
dtypes: float64(10), object(3)
memory usage: 60.6+ KB


In [26]:
# Verificando quantos NaN consta nas colunas 

dados_nan = afd_eja [["GEO", "LOC", "DA", "EJAFUNG1", "EJAFUNG2", "EJAFUNG3", "EJAFUNG4", "EJAFUNG5", "EJAMED1", "EJAMED2", "EJAMED3", "EJAMED4", "EJAMED5"]].isnull().sum()

print(dados_nan)

GEO           7
LOC           7
DA            7
EJAFUNG1    106
EJAFUNG2    106
EJAFUNG3    106
EJAFUNG4    106
EJAFUNG5    106
EJAMED1      87
EJAMED2      87
EJAMED3      87
EJAMED4      87
EJAMED5      87
dtype: int64


In [27]:
# Variáveis Numéricas

round(afd_eja.describe(), 2)

EJAFUNG1  EJAFUNG2  EJAFUNG3  EJAFUNG4  EJAFUNG5  EJAMED1  EJAMED2  \
count    489.00    489.00    489.00    489.00    489.00   508.00   508.00   
mean      34.28      0.69     47.90      2.71     14.42    55.48     1.99   
std       20.12      1.03     15.44      5.18     16.06    20.44     3.22   
min        0.00      0.00      0.00      0.00      0.00     0.00     0.00   
25%       18.00      0.10     40.50      1.40      2.60    39.30     0.30   
50%       34.00      0.40     47.90      2.10      7.80    54.45     0.80   
75%       47.20      0.90     56.90      2.80     20.40    71.25     2.00   
max      100.00      7.60    100.00    100.00     69.60   100.00    20.00   

       EJAMED3  EJAMED4  EJAMED5  
count   508.00   508.00   508.00  
mean     34.33     3.42     4.78  
std      18.48     6.76     7.33  
min       0.00     0.00     0.00  
25%      21.65     1.10     0.50  
50%      36.00     2.30     2.20  
75%      48.30     3.80     6.62  
max      80.00   100.00    65.20

In [28]:
# Variáveis Categóricas 

afd_eja.describe(include = [object])

GEO    LOC     DA
count      588    588    588
unique      33      3      6
top     Brasil  Total  Total
freq        18    197     99

In [29]:
afd_eja[["GEO", "LOC", "DA", "EJAFUNG1", "EJAFUNG2", "EJAFUNG3", "EJAFUNG4", "EJAFUNG5", "EJAMED1", "EJAMED2", "EJAMED3", "EJAMED4", "EJAMED5"]].describe()

EJAFUNG1    EJAFUNG2    EJAFUNG3    EJAFUNG4    EJAFUNG5     EJAMED1  \
count  489.000000  489.000000  489.000000  489.000000  489.000000  508.000000   
mean    34.281186    0.688344   47.903681    2.710429   14.416360   55.481496   
std     20.116210    1.025198   15.438282    5.182734   16.060954   20.437701   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%     18.000000    0.100000   40.500000    1.400000    2.600000   39.300000   
50%     34.000000    0.400000   47.900000    2.100000    7.800000   54.450000   
75%     47.200000    0.900000   56.900000    2.800000   20.400000   71.250000   
max    100.000000    7.600000  100.000000  100.000000   69.600000  100.000000   

          EJAMED2     EJAMED3     EJAMED4     EJAMED5  
count  508.000000  508.000000  508.000000  508.000000  
mean     1.990945   34.330512    3.421457    4.775591  
std      3.223936   18.475951    6.761178    7.334504  
min      0.000000    0.000000    0.000000    0.000000  
25%      0.300000   21.650000    1.100000    0.500000  
50%      0.800000   36.000000    2.300000    2.200000  
75%      2.000000   48.300000    3.800000    6.625000  
max     20.000000   80.000000  100.000000   65.200000

In [31]:
# Substituição dos nan pela mediana das colunas
afd_eja[col_num] = afd_eja[col_num].apply(lambda x: x.fillna(x.median()))

C:\Users\ruthm\AppData\Local\Temp\ipykernel_19492\4017561424.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  afd_eja[col_num] = afd_eja[col_num].apply(lambda x: x.fillna(x.median()))


**Verificando se o replace dos nan pela mediana das colunas funcionou**

In [32]:
afd_eja[col_num].isnull().sum()

EJAFUNG1    0
EJAFUNG2    0
EJAFUNG3    0
EJAFUNG4    0
EJAFUNG5    0
EJAMED1     0
EJAMED2     0
EJAMED3     0
EJAMED4     0
EJAMED5     0
dtype: int64

In [34]:
afd_eja.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   GEO       588 non-null    object 
 1   LOC       588 non-null    object 
 2   DA        588 non-null    object 
 3   EJAFUNG1  595 non-null    float64
 4   EJAFUNG2  595 non-null    float64
 5   EJAFUNG3  595 non-null    float64
 6   EJAFUNG4  595 non-null    float64
 7   EJAFUNG5  595 non-null    float64
 8   EJAMED1   595 non-null    float64
 9   EJAMED2   595 non-null    float64
 10  EJAMED3   595 non-null    float64
 11  EJAMED4   595 non-null    float64
 12  EJAMED5   595 non-null    float64
dtypes: float64(10), object(3)
memory usage: 60.6+ KB


### Visualizando os dados - EJA

*1. Em qual zona a EJA sofre uma maior discrepância entre a formação docente e a disciplina ensinada num total?*

*2. Analisando separadamente, qual zona apresenta maior discrepância na formação docente para a EJA do Ensino Fundamental? E para a EJA do Ensino Médio?*

*3. Em quais dependências admnistritativas há uma maior incidênica dessa discrepância em relação a EJA do Ensino Fundamental? E no Ensino Médio?*

*4. Há alguma relação entre a unidade geográfica e a localização em realação a EJA do Ensino Fundamental? E no Ensino Médio?*

*5. Existe alguma correlação entre a unidade geográfica e a dependencia administrativa em relação à formação docente da EJA no Ensino Fundamental? E no Ensino Médio?*

*6. Ao fazer uma comparação entre o Ensino Fundamental EJA e Ensino Médio EJA, qual podemos observar que há uma maior dispariadade?*

*7. Qual região do Brasil apresenta a maior variação na formação dos docentes da EJA em termos de adequação (Grupos 1 a 5)?*

*8. Qual grupo de formação docente (1, 2, 3, 4 e 5) concentra um maior percentual desses docentes, considerando a zona geográfica?*

*9. Qual grupo de formação docente (Grupo 1, 2, 3, 4, 5) apresenta maior percentual de docentes na EJA, considerando a dependência administrativa?*

*10. Comparando as regiões do Brasil, qual apresenta maior desigualdade na adequação da formação docente para a EJA?*

### Conclusão